app/src/main/java/org/wikipedia/views/ObservableWebView.kt

```diff
@@ -0,0 +1,209 @@
+                    totalAmountScrolled = 0
+                }
+                touchStartX = event.x
+                touchStartY = event.y
+                drawEventsWhileSwiping = 0
+            }
+            MotionEvent.ACTION_UP -> {
+                if (abs(event.x - touchStartX) <= touchSlop &&
+                        abs(event.y - touchStartY) <= touchSlop) {
+                    if (onClickListeners.any { it.onClick(event.x, event.y) }) {
+                        return true
+                    }
+                }
+                drawEventsWhileSwiping = 0
+                onUpOrCancelMotionEventListeners.forEach {
+                    it.onUpOrCancelMotionEvent()
+                }
+                drawEventsWhileSwiping = 0
+            }
+            MotionEvent.ACTION_CANCEL -> {
+                onUpOrCancelMotionEventListeners.forEach {
+                    it.onUpOrCancelMotionEvent()
+                }
+                drawEventsWhileSwiping = 0
+            }
+            else -> { }
+        }
+        return super.onTouchEvent(event)
+    }
+
+    private fun handleMouseRightClick(x: Float, y: Float) {
+        val eventTimeTravelMillis = 1000
+        post {
+            dispatchTouchEvent(MotionEvent.obtain(SystemClock.uptimeMillis(),
+                    SystemClock.uptimeMillis() + eventTimeTravelMillis,
+                    MotionEvent.ACTION_DOWN, x, y, 0))
+        }
+    }
+
+    override fun onDraw(canvas: Canvas) {
+        super.onDraw(canvas)
+        if (isInEditMode) {
+            return
+        }
+        drawEventsWhileSwiping++
+        if (drawEventsWhileSwiping > SWIPE_DRAW_TOLERANCE) {
+            setChildViewScrolled()
+        }
+        if (currentContentHeight != contentHeight) {
+            currentContentHeight = contentHeight
+            onContentHeightChangedListeners.forEach {
+                it.onContentHeightChanged(currentContentHeight)
+            }
+        }
+        WikipediaApp.getInstance().bus.post(INVALIDATE_EVENT)
+    }
+
+    companion object {
+        private val INVALIDATE_EVENT = WebViewInvalidateEvent()
+
+        /**
+         * Threshold (in pixels) of continuous scrolling, to be considered "fast" scrolling.
+         */
+        private val FAST_SCROLL_THRESHOLD = (1000 * densityScalar).toInt()
+
+        /**
+         * Maximum single scroll amount (in pixels) to be considered a "human" scroll.
+         * Otherwise it's probably a programmatic scroll, which we won't count.
+         */
+        private val MAX_HUMAN_SCROLL = (500 * densityScalar).toInt()
+
+        /**
+         * Maximum amount of time that needs to elapse before the previous scroll amount
+         * is "forgotten." That is, if the user scrolls once, then scrolls again within this
+         * time, then the two scroll actions will be added together as one, and counted towards
+         * a possible "fast" scroll.
+         */
```

Comment:
Can we remove all unnecessary comments since we have committed to self-commenting naming conventions?

Reply:
Done.